framework/src/base/FEProblem.C

```diff
@@ -1494,6 +1491,15 @@ FEProblem::addAuxKernel(const std::string & kernel_name, const std::string & nam
     parameters.set<SystemBase *>("_sys") = &_aux;
     parameters.set<SystemBase *>("_nl_sys") = &_nl;
   }
+
+  if (_displaced_problem != NULL)
+  {
+    if (!parameters.get<std::vector<BoundaryName> >("boundary").empty())
+      _reinit_displaced_face = true;
```

Comment:
To be clear: I'm not asking you to do anything.  I think this fix is "right" as far as what you can do.  The @idaholab/moose-team needs to work on this...